lisa/sut_orchestrator/azure/transformers.py

```diff
@@ -0,0 +1,265 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass, field
+from pathlib import PurePosixPath
+from time import sleep
+from typing import Any, Dict, List, Type
+
+from azure.mgmt.compute.models import GrantAccessData  # type: ignore
+from dataclasses_json import dataclass_json
+from retry import retry
+
+from lisa import schema
+from lisa.feature import Features
+from lisa.features import StartStop
+from lisa.node import RemoteNode
+from lisa.platform_ import load_platform
+from lisa.transformer import Transformer
+from lisa.util import LisaException, constants, get_date_str, get_datetime_path
+from lisa.util.perf_timer import create_timer
+
+from .common import (
+    check_or_create_storage_account,
+    get_compute_client,
+    get_network_client,
+    get_node_context,
+    get_or_create_storage_container,
+    get_storage_account_name,
+    wait_operation,
+)
+from .platform_ import AzurePlatform
+from .tools import Waagent
+
+DEFAULT_VHD_CONTAINER_NAME = "lisa-vhd-cache"
+DEFAULT_VHD_SUBFIX = "exported"
+
+
+@retry(tries=10, jitter=(1, 2))
+def _generate_vhd_path(container_client: Any, file_name_part: str = "") -> str:
+    path = PurePosixPath(
+        f"{get_date_str()}/{get_datetime_path()}_"
+        f"{DEFAULT_VHD_SUBFIX}_{file_name_part}.vhd"
+    )
+    blobs = container_client.list_blobs(name_starts_with=path)
+    for _ in blobs:
+        raise LisaException(f"blob exists already: {path}")
+    return str(path)
+
+
+@dataclass_json
+@dataclass
+class VhdTransformerSchema(schema.Transformer):
+    # resource group and vm name to be exported
+    resource_group_name: str = field(
+        default="", metadata=schema.metadata(required=True)
+    )
+    vm_name: str = "node-0"
+
+    # values for SSH connection. public_address is optional, because it can be
+    # retrieved from vm_name. The private_key_file must be provided. Others can
+    # use default values.
+    public_address: str = ""
+    public_port: int = 22
+    username: str = "lisatest"
```

Comment:
Are you sure you want to commit this?

Reply:
yes, the LISA is a test framework, it's the default user name, and be used in azure deployment too. Will add a constant to reuse it.